packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActionButton,
+  DialogActions,
+  DialogCancelButton,
+  DialogContent,
+  DialogHeader,
+  Form,
+  FormInput,
+  Text,
+} from "@prismicio/editor-ui";
+import { useState } from "react";
+
+import {
+  addUIDField,
+  getUIDField,
+  getUIDFieldLabel,
+  updateUIDField,
+} from "@/domain/customType";
+import { useCustomTypeState } from "@/features/customTypes/customTypesBuilder/CustomTypeProvider";
+
+export function UIDEditor() {
+  const [isOpen, setOpen] = useState(false);
+  const { customType, setCustomType } = useCustomTypeState();
+  const field = getUIDField(customType);
+  const uidFieldLabel = getUIDFieldLabel(customType);
+  const [label, setLabel] = useState(uidFieldLabel ?? "");
+
+  function handleSubmit() {
+    if (!label) {
+      return;
+    }
+    const updatedCustomType = field
+      ? updateUIDField(label, customType)
+      : addUIDField(label, customType);
+
+    setCustomType(updatedCustomType);
+    setOpen(false);
+  }
+
+  return (
+    <Dialog
+      open={isOpen}
+      onOpenChange={setOpen}
+      size="small"
+      trigger={
+        <Button
+          color="grey"
+          textColor="placeholder"
+          startIcon={field ? "language" : "add"}
+          sx={{ marginInline: "auto" }}
+        >
+          {field ? uidFieldLabel : "Add an UID"}
+        </Button>
+      }
+    >
+      <DialogHeader title="Edit the UID label" />
+      <DialogContent>
+        <Form onSubmit={handleSubmit}>
+          <Box flexDirection="column" padding={16} gap={4}>
+            <FormInput
+              type="text"
+              label="Label *"
+              placeholder="UID"
+              value={label}
+              onValueChange={setLabel}
+              error={!label && "This field is required"}
+            />
+            {label && <Text color="grey11">A label for the UID</Text>}
+          </Box>
+          <DialogActions>
+            <DialogCancelButton size="medium" />
```

Comment:
⚠️ #issue

Potential issue, when you click on "Cancel" or just close the Modal with the icon or ESC it will keep the value inside the field, looks strange as you want to cancel your action and I would have expected to have back the original value when opening the modal again no? Is this desired?

Reply:
Wasn't 100% sure what would be the best option. 
But you're right, reseting the field on exiting modal is more expected behaviour. 
Will change that.